source/common/filter/config_discovery_impl.h

```diff
@@ -169,7 +169,8 @@ class HttpDynamicFilterConfigProviderImpl
                                       ProtobufTypes::MessagePtr&& default_config,
                                       bool last_filter_in_filter_chain,
                                       const std::string& filter_chain_type,
-                                      absl::string_view stat_prefix)
+                                      absl::string_view stat_prefix,
+                                      const Network::ListenerFilterMatcherSharedPtr&)
```

Comment:
This is strange for adding a listener filter matcher parameter for the `HttpDynamicFilterConfigProviderImpl`.

Is that possible we pass the proto config inside the `FilterConfigProviderManager::createDynamicFilterConfigProvider` method, then the each derived class can rewrite its own version. For the TCP listener filter config provider, it can parse the proto config and create the listener filter matcher.

Reply:
Thanks for the comments.

Yes, the parameter list of this function is a little bit combination of different filters.  We will clean this up with item 8 of issue #20049. Not with this PR.